### PR TITLE
Write lookup segments

### PR DESCRIPTION
**Pull request by AlexeyRaga**

### CODE REVIEW

File: src/Arbor/File/Format/Asif/Write.hs [hunk 1]
```diff
@@ -244,6 +250,79 @@ utcTimeMicrosSegment f = genericFold BB.int64LE (Known F.TimeMicros64LE) (fromTi
...
+          let (v, c', m') = updateMap b c m
+          liftIO $ BB.hPutBuilder h $ enc v
+          pure (h, m', c')
+
+    lextract (h, _, _) = pure
+      [ segment h $ metaFilename name <> metaFormat fmt
+      ]
+
+    updateMap k c m =
+      maybe (c, c+1, Map.insert k c m) (, c, m) (Map.lookup k m)
```
Comment (newhoggy) [inline src/Arbor/File/Format/Asif/Write.hs]: More type signatures please 😁

File: src/Arbor/File/Format/Asif/Write.hs [hunk 2]
```diff
@@ -244,6 +250,79 @@ utcTimeMicrosSegment f = genericFold BB.int64LE (Known F.TimeMicros64LE) (fromTi
...
+-- @
+word16LookupSegment :: (MonadResource m, Ord b)
+  => T.Text
+  -> (a -> Maybe b)
+  -> FoldM m b [Segment Handle]
+  -> FoldM m a [Segment Handle]
+word16LookupSegment name f = lookupSegment name f (Known F.Word16LE) BB.word16LE
+
+-- | Creates a lookup segment where index keys are 'Word32'
+-- Missing values are represented by 'maxValue :: Word32'
```
Comment (dsturnbull) [inline src/Arbor/File/Format/Asif/Write.hs]: This should be maxBound I think

File: src/Arbor/File/Format/Asif/Write.hs (hunk 1 above)
Comment (AlexeyRaga) [reply src/Arbor/File/Format/Asif/Write.hs]: What do you mean? Adding type signature for `updateMap` in `where`?
Do you think that `(Num a, Ord k) => k -> a -> Map.Map k a -> (a, a, Map.Map k a)` would help? :)

File: src/Arbor/File/Format/Asif/Write.hs (hunk 2 above)
Comment (AlexeyRaga) [reply src/Arbor/File/Format/Asif/Write.hs]: Thanks!

**Comment by newhoggy** [comment]
👍

File: src/Arbor/File/Format/Asif/Write.hs (hunk 1 above)
Comment (newhoggy) [reply src/Arbor/File/Format/Asif/Write.hs]: Something like that.  Although consider using type variables from `lookupSegment` with scoped type variables.

Comment (dsturnbull) [reply src/Arbor/File/Format/Asif/Write.hs]: Or names

**Pull request merged**